assets/js/modules/idea-hub/components/dashboard/DashboardIdeasWidget/Idea.js

```diff
@@ -49,23 +50,33 @@ import UnpinIcon from '../../../../../../svg/idea-hub-unpin.svg';
 const { useDispatch } = Data;
 
 const Idea = ( { postEditURL, name, text, topics, buttons } ) => {
-	const { createIdeaDraftPost } = useDispatch( STORE_NAME );
+	const {
+		createIdeaDraftPost,
+		saveIdea,
+		unsaveIdea,
+		dismissIdea,
+	} = useDispatch( STORE_NAME );
 	const [ isProcessing, setIsProcessing ] = useState( false );
 
-	const handleDelete = useCallback( () => {
-		// @TODO: Implement callback.
-		global.console.log( `Deleted: ${ name }` );
-	}, [ name ] );
-
-	const handlePin = useCallback( () => {
-		// @TODO: Implement callback.
-		global.console.log( `Pinned: ${ name }` );
-	}, [ name ] );
-
-	const handleUnpin = useCallback( () => {
-		// @TODO: Implement callback.
-		global.console.log( `Unpinned: ${ name }` );
-	}, [ name ] );
+	const refreshNewSavedIdeas = useCallback( async () => {
+		await API.invalidateCache( 'modules', 'idea-hub', 'new-ideas' );
+		await API.invalidateCache( 'modules', 'idea-hub', 'saved-ideas' );
+	}, [] );
```

Comment:
Let's better move it to the `controlCallback` function of the `fetchPostUpdateIdeaStateStore` fetch store. We need to do the following:

1. Update the `controlCallback` function to return a new promise.
1. In the promise callback:
   1. Call and wait for results of the `API.set(...)` function.
   1. Call invalidate cache functions
   1. Resolve the promise passing the results of the `API.set(...)` function call.

Reply:
Updated @eugene-manuilov Can you check?

Reply:
Yes, looks good to me. Thanks!